TeachingRecordSystem/tests/TeachingRecordSystem.Core.Tests/Services/TrsDataSync/TrsDataSyncHelperTests.cs

```diff
@@ -37,4 +41,90 @@ public TrsDataSyncHelperTests(
     private TestableClock Clock { get; }
 
     public TrsDataSyncHelper Helper { get; }
+
+    private async Task<T> CreateDeactivatedEntityVersion<T>(
+        T existingEntity,
+        AuditDetailCollection auditDetailCollection)
+        where T : Entity
+    {
+        if (existingEntity.GetAttributeValue<OptionSetValue>("statecode").Value != 0)
+        {
+            throw new ArgumentException("Entity must be active.", nameof(existingEntity));
+        }
+
+        var currentDqtUser = await TestData.GetCurrentCrmUser();
+
+        var updatedEntity = existingEntity.Clone();
+        updatedEntity.Attributes["statecode"] = new OptionSetValue(1);
+        updatedEntity.Attributes["statuscode"] = new OptionSetValue(2);
+
+        var oldValue = new Entity(dfeta_qualification.EntityLogicalName, existingEntity.Id);
```

Comment:
Should the entity name be passed into this method if it is also used for alerts?